### PR TITLE
Remove Class.isPrimitiveClass and outdated methods

### PR DESCRIPTION
**Pull request by theresa-m**
Related to https://github.com/eclipse-openj9/openj9/issues/18157

Closes: https://github.com/eclipse-openj9/openj9/issues/13615

### CODE REVIEW

File: jcl/src/java.base/share/classes/java/lang/Class.java [hunk 1]
```diff
@@ -2631,14 +2631,6 @@ public boolean isInterface() {
 public native boolean isPrimitive();
 
 /*[IF INLINE-TYPES]*/
-/**
- * Answers true if the receiver represents a primitive class type. Array classes
- * return false.
- *
- * @return	true if receiver is primitive class type, and false otherwise.
- */
-native boolean isPrimitiveClass();
```
Comment (hangshao0) [inline jcl/src/java.base/share/classes/java/lang/Class.java:2640]: There are references of this method in JIT code like: https://github.com/eclipse-openj9/openj9/blob/6036c034f10ce416854c5182b861c7c38d431661/runtime/compiler/codegen/J9RecognizedMethodsEnum.hpp#L62

Comment (theresa-m) [reply jcl/src/java.base/share/classes/java/lang/Class.java:2640]: Thanks. I removed the remaining references.

**Comment by hangshao0** [comment]
There are code changes in JIT here related to the removal of `Class.isPrimitiveClass()`. Adding @hzongaro and @a7ehuo as reviewers.

**Comment by a7ehuo** [comment]
The related change in the JIT under `runtime/compiler` for the removal of `java_lang_Class_isPrimitiveClass` looks good to me. Thank you very much for the update @theresa-m!

**Review by hzongaro — APPROVED** [review]
The JIT changes look good to me.  Thanks!

**Comment by hangshao0** [comment]
@theresa-m There is a merge conflict.

**Comment by theresa-m** [comment]
I fixed the merge conflict.

**Comment by hangshao0** [comment]
Jenkins test sanity,extended zlinuxval jdknext

**Comment by hangshao0** [comment]
Jenkins test sanity,extended amac jdk21

**Review by hangshao0 — APPROVED** [review]

**Pull request merged**